### PR TITLE
Add all back

### PR DESCRIPTION
**Pull request by maxandron**

### CODE REVIEW

File: color_coding_script.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    with open(full_path('color_coding_settings.json')) as f:
+        try:
+            data = json.load(f)
+        except:
+            print('The JSON File is corrupted, modify the file and re-run the script')
+            sys.exit()
+
+    CALENDER_ID = data["calender_id"]
+    
+    store = file.Storage(full_path('token.json'))
```
Comment (maxandron) [inline color_coding_script.py]: file is a reserved keyword

File: color_coding_script.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+from __future__ import print_function
+import datetime
+from googleapiclient.discovery import build
+from httplib2 import Http
+from oauth2client import file, client, tools
```
Comment (maxandron) [inline color_coding_script.py]: import file as a different name
import <> from <> as <>

File: color_coding_script.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        print(counter, 'Events were updated successfully!')
+
+
+def full_path(file_name):
+    dir_path = os.path.dirname(__file__)
+    return os.path.join(dir_path, file_name)
+
+
+def main():
+    with open(full_path('color_coding_settings.json')) as f:
```
Comment (maxandron) [inline color_coding_script.py]: move to a retrieve_settings function

File: color_coding_script.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            sys.exit()
+
+    CALENDER_ID = data["calender_id"]
+    
+    store = file.Storage(full_path('token.json'))
+    creds = store.get()
+    if not creds or creds.invalid:
+        flow = client.flow_from_clientsecrets(full_path('credentials.json'), SCOPES)
+        creds = tools.run_flow(flow, store)
+    service = build('calendar', 'v3', http=creds.authorize(Http()))
```
Comment (maxandron) [inline color_coding_script.py]: move to a different calendar_service function

File: color_coding_script.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    # Call the Calendar API
+    now = datetime.datetime.utcnow().isoformat() + 'Z' # 'Z' indicates UTC time
+    events_result = service.events().list(calendarId=CALENDER_ID, timeMin=now,
+                                        maxResults=200, singleEvents=True,
+                                        orderBy='startTime').execute()
+    events = events_result.get('items', [])
+    colors = service.colors().get().execute()
+    found = 0
+    counter = 0
+    colors_dict = {'BLUE' : 1,
```
Comment (maxandron) [inline color_coding_script.py]: Move to COLORS_DICT constant

File: color_coding_script.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    service = build('calendar', 'v3', http=creds.authorize(Http()))
+    
+    # Call the Calendar API
+    now = datetime.datetime.utcnow().isoformat() + 'Z' # 'Z' indicates UTC time
+    events_result = service.events().list(calendarId=CALENDER_ID, timeMin=now,
+                                        maxResults=200, singleEvents=True,
+                                        orderBy='startTime').execute()
+    events = events_result.get('items', [])
+    colors = service.colors().get().execute()
+    found = 0
```
Comment (maxandron) [inline color_coding_script.py]: you initialize found later, this initialization is redundant

File: color_coding_script.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        print('No upcoming events found.')
+        sys.exit()
+        
+    for event in events:
+        found = 0
+        EVENT_ID = event['id']
+        
+        try:
+            summary = event[data["tag_location"]] 
+        except:
```
Comment (maxandron) [inline color_coding_script.py]: never except with nothing. catch the specific exception
in this case I guess you are scared of KeyError

you can avoid catching an exception alltogether by using the get function
`event.get(data["tag_location"], '')`

File: color_coding_script.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            summary = ''
+        
+        if summary != '':
```
Comment (maxandron) [inline color_coding_script.py]: just `if summary`

File: color_coding_script.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                    'BOLD_BLUE' : 9,
+                    'BOLD_GREEN' : 10,
+                    'BOLD_RED' : 11}
+    
+    if not events:
+        print('No upcoming events found.')
+        sys.exit()
+        
+    for event in events:
+        found = 0
```
Comment (maxandron) [inline color_coding_script.py]: move the whole for content into a separate function

File: color_coding_script.py [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+# If modifying these scopes, delete the file token.json.
+SCOPES = 'https://www.googleapis.com/auth/calendar' 
+
+def print_result(counter):
+    if counter == 0:
+        print('no events were updated')
+    elif counter == 1:
+        print(counter, 'Event was updated successfully!')
+    else:
+        print(counter, 'Events were updated successfully!')
```
Comment (maxandron) [inline color_coding_script.py]: I want to know how many events were updates. use string formatting

File: color_coding_script.py [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+import os
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = 'https://www.googleapis.com/auth/calendar' 
+
+def print_result(counter):
+    if counter == 0:
+        print('no events were updated')
+    elif counter == 1:
+        print(counter, 'Event was updated successfully!')
```
Comment (maxandron) [inline color_coding_script.py]: Change to "One event was"
from python zen: https://www.python.org/dev/peps/pep-0020/
`Explicit is better than implicit.`

File: color_coding_settings.json [hunk 12]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (maxandron) [inline color_coding_settings.json]: change file name to just settings.json

**Comment by DanaGriff** [comment]
i think i should remove credentials.json from gitignore. what do you think?
cause the user doesnt have to access google api and regenerate one for himself.

File: .gitignore [hunk 13]
```diff
@@ -104,3 +104,4 @@ venv.bak/
 .mypy_cache/
 
 credentials.json
```
Comment (maxandron) [inline .gitignore]: reminder to remove credentials

File: color_coding_script.py [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+from oauth2client import client, tools
+import json
+import sys
+from pprint import pprint
+import os
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = 'https://www.googleapis.com/auth/calendar' 
+COLORS = {'BLUE' : 1,
+                    'GREEN' : 2,
```
Comment (maxandron) [inline color_coding_script.py]: align this

File: color_coding_script.py [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+                    'BOLD_GREEN' : 10,
+                    'BOLD_RED' : 11}
+ 
+def color_event(event):
+    found = 0
+    EVENT_ID = event['id']
+    
+    summary = event.get('summary', '')
+    description = event.get('description', '')
+
```
Comment (maxandron) [inline color_coding_script.py]: why not just joining summary and description? could save you some code

File: color_coding_script.py [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+    found = 0
+    EVENT_ID = event['id']
+    
+    summary = event.get('summary', '')
+    description = event.get('description', '')
+    
+    if summary or description:
+        for color in data["color_coding"]:
+            if color["tag"] in summary or color["tag"] in description:
+                EVENT={'colorId': COLORS[color["color"]]}
```
Comment (maxandron) [inline color_coding_script.py]: EVENT is not a constant, hence should be lower cased

File: color_coding_script.py [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+                found = 1
+                break
+                
+        if found != 0:
```
Comment (maxandron) [inline color_coding_script.py]: why use 0 and 1 and not True/False ?

File: color_coding_script.py [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+def full_path(file_name):
+    dir_path = os.path.dirname(__file__)
+    return os.path.join(dir_path, file_name)
+
+def retrieve_settings():
+    with open(full_path('settings.json')) as f:
+        try:
+            json_data = json.load(f);
+            return json_data;
+        except:
```
Comment (maxandron) [inline color_coding_script.py]: Never write `except:`.
For a longer explanation: https://stackoverflow.com/questions/21553327/why-is-except-pass-a-bad-programming-practice
which error are you afraid of here? ValueError? then catch this

File: color_coding_script.py [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+            print 'The JSON File is corrupted, modify the file and re-run the script'
+            sys.exit()
+
+def main():
+    data = retrieve_settings()
+    CALENDER_ID = data["calender_id"]
+    
+    store = my_file.Storage(full_path('token.json'))
```
Comment (maxandron) [inline color_coding_script.py]: lines 68-72 can be a different function

Comment (maxandron) [reply color_coding_script.py]: that returns a service

File: color_coding_script.py [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+    service = build('calendar', 'v3', http=creds.authorize(Http())) # should i do some exception handling here?
+    
+    # Call the Calendar API
+    now = datetime.datetime.utcnow().isoformat() + 'Z' # 'Z' indicates UTC time
+    
+    events_result = service.events().list(calendarId=CALENDER_ID, timeMin=now,
+                                        maxResults=200, singleEvents=True,
+                                        orderBy='startTime').execute()
+                                        
+    print events_result
```
Comment (maxandron) [inline color_coding_script.py]: leftover prints

File: settings.json [hunk 21]
```diff
@@ -0,0 +1,33 @@
+{
+    "calender_id": "<insert_id>",
+	"color_coding": [
```
Comment (maxandron) [inline settings.json]: fucked up indent

**Pull request merged**